src/test/java/net/sf/jabref/importer/fileformat/BiblioscapeImporterTest.java

```diff
@@ -0,0 +1,156 @@
+        try (InputStream is = BiblioscapeImporter.class.getResourceAsStream("BiblioscapeImporterTest2.txt")) {
+            Assert.assertEquals(null, importer.importEntries(is, new OutputPrinterToNull()));
+        }
+    }
+
+    @Test
+    public void testImportEntries3() throws Throwable {
+        BiblioscapeImporter importer = new BiblioscapeImporter();
+        BibtexImporter bibtexImporter = new BibtexImporter();
+
+        try (InputStream biblioStream = BiblioscapeImporter.class.getResourceAsStream("BiblioscapeImporterTest3.txt");
+                InputStream bibStream = BiblioscapeImporter.class
+                        .getResourceAsStream("BiblioscapeImporterTest3_comparefile.bib")) {
+
+            List<BibEntry> isEntries = importer.importEntries(biblioStream, new OutputPrinterToNull());
+            List<BibEntry> shouldBeEntries = bibtexImporter.importEntries(bibStream, new OutputPrinterToNull());
+            Assert.assertEquals(shouldBeEntries.size(), isEntries.size());
+
+            int c = 0;
+            for (int i = 0; i < isEntries.size(); i++) {
+                for (int j = 0; j < shouldBeEntries.size(); j++) {
+                    BibEntry isEntry = isEntries.get(i);
+                    BibEntry shouldBeEntry = shouldBeEntries.get(j);
+                    if (isEntry.getField("isbn").equals(shouldBeEntry.getField("isbn"))) {
+                        BibtexEntryAssert.assertEquals(shouldBeEntry, isEntry);
+                        c++;
+                    }
+                }
+            }
+            Assert.assertEquals(isEntries.size(), c);
+
+
+        }
+    }
+
+    @Test
+    public void testImportEntries4() throws Throwable {
+        BiblioscapeImporter importer = new BiblioscapeImporter();
+        BibtexImporter bibtexImporter = new BibtexImporter();
+
+        try (InputStream biblioStream = BiblioscapeImporter.class.getResourceAsStream("BiblioscapeImporterTest4.txt");
+                InputStream bibStream = BiblioscapeImporter.class
+                        .getResourceAsStream("BiblioscapeImporterTest4_comparefile.bib")) {
+
+            List<BibEntry> isEntries = importer.importEntries(biblioStream, new OutputPrinterToNull());
+            List<BibEntry> shouldBeEntries = bibtexImporter.importEntries(bibStream, new OutputPrinterToNull());
+            Assert.assertEquals(shouldBeEntries.size(), isEntries.size());
+
+            int c = 0;
+            for (int i = 0; i < isEntries.size(); i++) {
+                for (int j = 0; j < shouldBeEntries.size(); j++) {
+                    BibEntry isEntry = isEntries.get(i);
+                    BibEntry shouldBeEntry = shouldBeEntries.get(j);
+                    if (isEntry.getField("note").equals(shouldBeEntry.getField("note"))) {
+                        BibtexEntryAssert.assertEquals(shouldBeEntry, isEntry);
+                        c++;
+                    }
+                }
+            }
```

Comment:
@simonharrer I don't know why but the `BibtexImporter` does not return the Bibentries from the `List` in same order as they are saved in origin bib file. But `BiblioscapeImporter` does. Therefore I inserted a sort of ID for each entry, so that I can compare correctly (line 144). To compare  them manually will blow up the method, cause there are at least 10 entries.